{{cookiecutter.repo_name}}/devtools/create_dev_env.py

```diff
@@ -0,0 +1,128 @@
+import os
+import argparse
+from pathlib import Path
+from conda.common.serialize import yaml_load
+from conda_env.pip_util import pip_args
+import subprocess as sp
+try:
+    from conda_build import api
+    from conda_build.environ import create_env
+    from conda_build.config import Config
+    from conda_build.conda_interface import url_path
+except ImportError as e:
+    e.msg += "\nNote:\n"
+    e.msg += "-------\n"
+    e.msg += "\nYou need to make sure that 'conda-build' is installed:"
+    e.msg += "\n    hint ---> `conda install conda-build`"
+    raise
+
+
+def ask_yn():
+    def question():
+        prompt = "Continue? [Y/n]"
+        try:
+            return raw_input(prompt).lower() or 'y'
+        except KeyboardInterrupt:
+            print('')
+            return 'n'
+        except NameError:
+            return input(prompt).lower() or 'y'
+
+    answer = question()
+    while not answer.startswith(('y', 'n')):
+        print("{} is not valid, please answer 'yes' or 'no'")
+        answer = question()
+    if answer.startswith('n'):
+        return False
+    else:
+        return True
+
+
+parser = argparse.ArgumentParser(
+    description=
+    "Creates a conda environment with the specified python version, and the run/test requirements of the provided recipe"
+)
+
+parser.add_argument(
+    "-n", "--name", type=str, help="The name of the environment, existing environments will be updated")
+parser.add_argument("-p", "--python", type=str, help="The version of python to be installed in environment")
+parser.add_argument(
+    "-c",
+    "--channel",
+    type=str,
+    nargs="+",
+    help="Additional channel(s) to search for packages (searched in order given, before defaults)")
+parser.add_argument("-y", "--yes", dest='ask', action='store_false')
+parser.add_argument("--file", type=str, help="An environment file (yaml) from which additional packages will be taken")
+parser.add_argument("recipe_path", type=str, help="The path to the recipe that will be used")
+
+parser.set_defaults(ask=True)
+
+
+def setup_dev_env(args):
+
+    name = None
+    channels = args.channel if args.channel else []
+    deps = []
+    pip_deps = []
+
+    if args.file:
+        args_yaml = yaml_load(Path(args.file).read_text())
+        name = args_yaml.get('name', name)
+        pip_deps = []
+        for dep in args_yaml.get('dependencies', []):
+            if isinstance(dep, type(args_yaml)):
+                pip_deps.extend(dep.get('pip', []))
+                deps.append('pip')
+            else:
+                deps.append(dep)
+        channels.extend(args_yaml.get('channels', []))
+
+    config = Config(python=args.python)
```

Comment:
This seems like an unsafe call since the `Config` is not part of the true API (correct me if I misunderstand @loriab ) and is subject to change, even between minor versions. I think a safer call would be to pass `channel_urls=` keyword arg to the `render()` call, since all additional `**kwargs` are passed through to the Config constructor anyways.

This construct also makes no assumptions about channel structure and will only search exactly the channels you give it. So if you want to always include the `defaults` channel as a fallback, you would need to add `defaults` to the end of your iterable. e.g. If `url` is empty, then then `config.channel_urls = []` and *no* channels will be searched, so no dependencies would be resolved.